File: engine/engine-model/src/main/scala/net/server/coordinator/matchchecker/MapleMatchCheckingEntry.scala
package net.server.coordinator.matchchecker

class MapleMatchCheckingEntry(private var _cid: Int) {
  def cid: Int = _cid

  private var _accepted: Boolean = false

  def accepted: Boolean = _accepted

  def accept(): Boolean = {
    if (!_accepted) {
      _accepted = true
      true
    } else {
      false
    }
  }
}

File: engine/engine-model/src/main/java/net/server/coordinator/matchchecker/MapleMatchCheckingEntry.java
package net.server.coordinator.matchchecker;

public class MapleMatchCheckingEntry {
   private final Integer cid;

   private Boolean accepted = false;

   public MapleMatchCheckingEntry(Integer cid) {
      this.cid = cid;
   }

   public Integer cid() {
      return cid;
   }

   public Boolean accepted() {
      return accepted;
   }

   public Boolean accept() {
      if (!accepted) {
         accepted = true;
         return true;
      }
      return false;
   }
}
